src/client.rs

```diff
@@ -245,6 +263,16 @@ impl Pusher{
     self._trigger(channel_strings, event, payload, None)
   }
 
+  pub fn _make_url(&self) -> Url {
+    let host = match self.cluster.clone() {
+        Some(ref cluster) => format!("api-{}.pusherapp.com", cluster),
+        None => self.host.to_string()
+    };
```

Comment:
In fact, I think you might find it easier to generate the host in the `finalize` method, and just use that instead of separate `host` and `cluster` fields going forward.